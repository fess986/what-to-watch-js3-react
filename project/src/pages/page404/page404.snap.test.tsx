import {render} from '@testing-library/react';
import { createMemoryHistory } from 'history';

import Page404 from './page404';
import HistoryRouter from '../../components/history-route/history-route'; // по какой то непонятной мне причине, jest вообще ничего знать не хочет об объекте истории браузера, но при этом отлично понимает роутинг

describe('testing Page404 render', () => {
  it('Should render correctly', () => {

    const history = createMemoryHistory();// создаем специальный объект истории в памяти.

    // делаем обертку в виде роутинга над нашим компонентом, чтобы в нем могли работать такие вещи, как Link
    const {container} = render(
      <HistoryRouter history={history}>
        <Page404 />
      </HistoryRouter>,
    );

    expect(container).toMatchSnapshot();

  });
});
