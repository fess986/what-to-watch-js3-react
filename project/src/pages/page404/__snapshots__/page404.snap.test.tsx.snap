// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Page404 render Should render correctly 1`] = `
<div>
  <div
    class="wrapper404"
  >
    <div
      class="mainbox"
    >
      <div
        class="err404"
      >
        4
      </div>
      <div
        class="err404-2"
      >
        0
      </div>
      <div
        class="err404-3"
      >
        4
      </div>
      <div
        class="msg404"
      >
        Maybe this page moved? Got deleted? Is hiding out in quarantine? Never existed in the first place?
        <p>
          Let
          '
          s go
          <a
            class="link404"
            href="/"
          >
            home
          </a>
          and try from there.
        </p>
      </div>
    </div>
  </div>
</div>
`;
